cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -821,8 +839,11 @@ func buildScaleUpStatusNodeGroup(isScaleUpInProgress bool, isSafeToScaleUp bool,
 	}
 	if isScaleUpInProgress {
 		condition.Status = api.ClusterAutoscalerInProgress
-	} else if !isSafeToScaleUp {
+	} else if !scaleUpSafety.Healthy {
+		condition.Status = api.ClusterAutoscalerUnhealthy
+	} else if !scaleUpSafety.SafeToScale {
 		condition.Status = api.ClusterAutoscalerBackoff
+		condition.Message = scaleUpSafety.BackoffStatus.ErrorInfo.ErrorMessage
```

Comment:
I wouldn't do this in this PR yet, can we hold off on actual status changes for the follow-up PR and keep this as a pure refactor? This would change the current status format, but AFAIU not in the intended way we're aiming for in #6318